decidim/decidim
# Identify in person voters

#### :tophat: What? Why?
This PR implements the identification and verification of voters for in person voting.

The navigation flow is not definitive, as @agustibr will refactor it in #7801, so it's still not possible to get the election where the person will vote (at the moment, we get one election from the polling station's Voting).

Also, this PR doesn't check if the person has already voted nor mark that the person has voted, those features will be included in future PRs.

Finally, this PR add the autojump feature to the date of birth fields in the online and in person voting forms. It sets the focus to the next field when trying to add a third number to the day or month fields. It also sets the focus to the previous field when pressing the delete key in an empty month or year field.

#### :pushpin: Related Issues
- Related to #7124 and #7705

#### Testing
1. Create a voting with at least one election, upload a census CSV, asign the admin user as a polling officer and go to the user's account page.
2. Enter to the Polling Officer zone and click on the user icon (Identify and verify a person) for a polling station.
3. Fill the form with the data for a person included in the census CSV and press the "VALIDATE DOCUMENT" button.
4. Verify the name of the person and press the "VERIFY DOCUMENT" button.
5. Review the election questions, check the checkbox with the label "The participant has voted" and press the "COMPLETE VOTING" button.

#### :clipboard: Checklist
:rotating_light: Please review the [guidelines for contributing](https://github.com/decidim/decidim/blob/develop/CONTRIBUTING.adoc) to this repository.


### :camera: Screenshots

https://user-images.githubusercontent.com/453545/114419980-32955c80-9bb4-11eb-96c2-0ba588270a06.mov

Autofocus:

https://user-images.githubusercontent.com/453545/114420734-dc74e900-9bb4-11eb-89ea-d08ad2b2c1e3.mov


:hearts: Thank you!
